contents/kotlin/NullSafety.md

```diff
@@ -0,0 +1,121 @@
+# Null Safety in Kotlin
+
+Author: [Pan Haozhe](https://github.com/Haozhe321)
+
+# Overview
+Another part of this book chapter gives an introduction to Kotlin. This piece aims to discuss the null safety feature in Kotlin, and what was previously missing in Java.
+
+# The Billion Dollar Mistake
+“I call it my billion-dollar mistake. It was the invention of the null reference…My goal was to ensure that all use of references should be absolutely safe, with **checking performed automatically by the compiler**. But I couldn't resist the temptation to put in a null reference, simply because it was so easy to implement.”  
+-Tony Hoare  
+
+
+# NullPointerException
+As we know, Kotlin is now supported as the official language of Android applications. When developing Android applications in Java, NullPointerException(NPE) was a big problem. In fact, About one third of app crashes can be attributed to NPE. To see how it happens, let's take a look at the piece of Java code below:
```

Comment:
Can consider putting a hyperlink at `NullPointerException (NPE)`

Reply:
Also, I feel like the first line "As we know, Kotlin is now supported as the official language of Android applications." should not be included in this subsection as this subsection is entirely about NPE in Java and how Java settles NPE. Instead, I see it fitting in better as the first statement of the next subsection (which is on Kotlin).

Reply:
I removed "As we know, Kotlin is now supported as the official language of Android applications." because it is not useful for the purpose of this book chapter. There's another one on it.